app/controllers/sitemap_controller.rb

```diff
@@ -0,0 +1,47 @@
+class SitemapController < ApplicationController
+
+  def sitemap
+    #if false
+      #redirect_to "/" # The CDN then fetches the sitemap.xml from the sitemap/generate URL which points to the `generate` action
+    #else
+    render_site_map
+    #end
+  end
+
+  def generate
+    render_site_map
+  end
+
+  private
+
+    def render_site_map
+      respond_to do |format|
+
+        data = Rails.cache.fetch(@current_community, expires_in: 24.hours) do
```

Comment:
To my understanding, if you use `@current_community` as a cache key, the cache will be invalidated if the marketplace admin does any change to the marketplace that changes the `@current_community.updated_at`, and that's probably unwanted behaviour. So I'd recommend to use the marketplace ID instead of the community object.

In addition to the markeplace ID, the cache key needs to have some prefix to prevent it from clashing. So something like this:

```ruby
Rails.cache.fetch("sitemap-#{@current_community.id}", expires_in: 24.hours) do ...
```

Does it make sense?